api/manifest.go

```diff
@@ -60,55 +62,74 @@ func getManifestForSHA(ctx context.Context, sha string) (manifest []byte, err er
 }
 
 func getGitHubReleaseAssetForSHA(client gitHubClient, sha string) (manifest []byte, err error) {
-	// Search for the PR associated with the SHA.
-	url := gitHubSHASearchURL(sha)
-	var body []byte
-	if body, err = client.fetch(url); err != nil {
-		return nil, err
-	}
+	var releaseBody []byte
+	var releaseTag string
+	if sha == "" || sha == "latest" {
+		// Use GitHub's API for latest release.
+		releaseTag = "lastest"
+		url := gitHubLatestReleaseURL
+		if releaseBody, err = client.fetch(url); err != nil {
+			return nil, err
+		}
+	} else {
+		// Search for the PR associated with the SHA.
+		url := gitHubSHASearchURL(sha)
+		var body []byte
+		if body, err = client.fetch(url); err != nil {
+			return nil, err
+		}
 
-	var queryResults map[string]*json.RawMessage
-	if err = json.Unmarshal(body, &queryResults); err != nil {
-		return nil, err
-	}
-	var issues []map[string]*json.RawMessage
-	if err = json.Unmarshal(*queryResults["items"], &issues); err != nil {
-		return nil, err
-	}
-	if len(issues) < 1 {
-		return nil, fmt.Errorf("No search results found for SHA %s", sha)
-	}
+		var queryResults map[string]*json.RawMessage
+		if err = json.Unmarshal(body, &queryResults); err != nil {
+			return nil, err
+		}
+		var issues []map[string]*json.RawMessage
+		if err = json.Unmarshal(*queryResults["items"], &issues); err != nil {
+			return nil, err
+		}
+		if len(issues) < 1 {
+			return nil, fmt.Errorf("No search results found for SHA %s", sha)
+		}
 
-	// Load the release by the presumed tag name merge_pr_*
-	var prNumber int
-	if err = json.Unmarshal(*issues[0]["number"], &prNumber); err != nil {
-		return nil, err
-	}
+		// Load the release by the presumed tag name merge_pr_*
+		var prNumber int
+		if err = json.Unmarshal(*issues[0]["number"], &prNumber); err != nil {
+			return nil, err
+		}
 
-	releaseTag := fmt.Sprintf("merge_pr_%d", prNumber)
-	url = gitHubReleaseURL(releaseTag)
-	if body, err = client.fetch(url); err != nil {
-		return nil, err
+		releaseTag = fmt.Sprintf("merge_pr_%d", prNumber)
+		url = gitHubReleaseURL(releaseTag)
+		if releaseBody, err = client.fetch(url); err != nil {
+			return nil, err
+		}
 	}
 
 	var release map[string]*json.RawMessage
-	if err = json.Unmarshal(body, &release); err != nil {
+	if err = json.Unmarshal(releaseBody, &release); err != nil {
 		return nil, err
 	}
+
 	var assets []map[string]*json.RawMessage
 	if err = json.Unmarshal(*release["assets"], &assets); err != nil {
 		return nil, err
 	}
 	if len(assets) < 1 {
-		return nil, fmt.Errorf("No assets found for release %s", releaseTag)
+		return nil, fmt.Errorf("No assets found for %s release", releaseTag)
 	}
 	// Get (and unzip) the asset with name "MANIFEST-{sha}.json.gz"
+	shaMatch := sha
+	if sha == "" || sha == "latest" {
+		shaMatch = "[0-9a-f]{1,40}"
```

Comment:
This doesn't matter much, but it seems like the shortest sha that git will even try to match on the command line is 4 characters, so if shorter things show up here it would be very surprising. Do you think it's too conservative to assume exactly 40 chars and otherwise not match?